src/prebuilt/wasm2c.include.c

```diff
@@ -121,25 +121,25 @@ const char SECTION_NAME(declarations)[] =
 "  : (UNLIKELY((x) == 0 && (y) == 0)) ? (signbit(x) ? y : x) \\\n"
 "  : (x > y) ? x : y)\n"
 "\n"
-"#define TRUNC_S(ut, st, ft, min, max, maxop, x)                             \\\n"
-"   ((UNLIKELY((x) != (x))) ? TRAP(INVALID_CONVERSION)                       \\\n"
-"  : (UNLIKELY((x) < (ft)(min) || (x) maxop (ft)(max))) ? TRAP(INT_OVERFLOW) \\\n"
-"  : (ut)(st)(x))\n"
-"\n"
-"#define I32_TRUNC_S_F32(x) TRUNC_S(u32, s32, f32, INT32_MIN, INT32_MAX, >=, x)\n"
-"#define I64_TRUNC_S_F32(x) TRUNC_S(u64, s64, f32, INT64_MIN, INT64_MAX, >=, x)\n"
-"#define I32_TRUNC_S_F64(x) TRUNC_S(u32, s32, f64, INT32_MIN, INT32_MAX, >,  x)\n"
-"#define I64_TRUNC_S_F64(x) TRUNC_S(u64, s64, f64, INT64_MIN, INT64_MAX, >=, x)\n"
-"\n"
-"#define TRUNC_U(ut, ft, max, maxop, x)                                    \\\n"
-"   ((UNLIKELY((x) != (x))) ? TRAP(INVALID_CONVERSION)                     \\\n"
-"  : (UNLIKELY((x) <= (ft)-1 || (x) maxop (ft)(max))) ? TRAP(INT_OVERFLOW) \\\n"
-"  : (ut)(x))\n"
-"\n"
-"#define I32_TRUNC_U_F32(x) TRUNC_U(u32, f32, UINT32_MAX, >=, x)\n"
-"#define I64_TRUNC_U_F32(x) TRUNC_U(u64, f32, UINT64_MAX, >=, x)\n"
-"#define I32_TRUNC_U_F64(x) TRUNC_U(u32, f64, UINT32_MAX, >,  x)\n"
-"#define I64_TRUNC_U_F64(x) TRUNC_U(u64, f64, UINT64_MAX, >=, x)\n"
+"#define TRUNC_S(ut, st, ft, min, minop, max, x)                             \\\n"
+"  ((UNLIKELY((x) != (x)))                        ? TRAP(INVALID_CONVERSION) \\\n"
+"   : (UNLIKELY(!((x)minop(min) && (x) < (max)))) ? TRAP(INT_OVERFLOW)       \\\n"
+"                                                 : (ut)(st)(x))\n"
+"\n"
+"#define I32_TRUNC_S_F32(x) TRUNC_S(u32, s32, f32, (f32)INT32_MIN, >=, 2147483648.f, x)\n"
```

Comment:
shouldn't the minimum be `> 2147483649.f`?

Reply:
Not for the f32 -> i32 conversion, I believe. 2147483649.f cannot be represented in f32.

Reply:
(oops, I meant negative before)

but `-2147483648.8` rounds up to a valid number, an i32 of `-2147483648`?

Reply:
See https://float.exposed/0xcf000000. The next number that can be represented is https://float.exposed/0xcf000001, which is -2147483904.0.

Reply:
I see, thanks!